### DISCUSSION TITLE
smart output capability

### DISCUSSION THREAD

**Discussion by ycat3** [s-JoL/Open-Llama#57] [Ideas]
I am writing a python code for CLI smart output, this means streaming output per each following token in almost real time.
Easy output halfway progress monitoring without patient.
Comparing the English and Japanese conversation result between chat_server.py and my_smartchat_cli.py,
it seems chat_server.py is better than my_smartchat_cli.py in conversation quality.
What's is your default and recommended settings for temperature, top_k, top_p, repetition_penalty, and other related parameters.

**Comment by s-JoL** [comment]
good idea! Using streaming output will significantly improve the user experience. I have not made any special adjustments to the generation parameters. Currently, the default parameters are used. For details, see [GenerationConfig](https://huggingface.co/docs/transformers/v4.29.1/en/main_classes/text_generation#transformers.GenerationConfig).

**Reply by s-JoL** [reply]
Since I set do_sample=True, some fluctuations in the performance are acceptable.

**Comment by ycat3** [comment]
Thank you.
How about separator and stop token ?
I am using separator=" " and "</s>", stop token = default EOS only.
Is it enough ?
Anyway I will PR my code until early next week since I go to travel this week.
I plan to include the code for multiple GPUs optimization as well.

**Reply by s-JoL** [reply]
You are welcome to participate in the development together. I'm not sure what the sep token refers to here? Is it the separator between multiple rounds of dialogue? I also use eos token for splitting multiturn dialogue.

**Comment by ycat3** [comment]
typo
I am using separator=" " and "</s>", stop token = default EOS only.

**Comment by ycat3** [comment]
typo
I am using separator=" " and "\</s>", stop token = default EOS only
